webserver/grpc/src/main/java/io/helidon/webserver/grpc/GrpcProtocolHandler.java

```diff
@@ -172,7 +168,8 @@ public Http2StreamState streamState() {
 
     @Override
     public void rstStream(Http2RstStream rstStream) {
-        listener.onComplete();
+        callCancelled = (rstStream.errorCode() == Http2ErrorCode.CANCEL);
```

Comment:
Question: I don't know anything about the protocol. It looks as if we create a new `GrpcProtocolHandler` for each incoming request (the parameter list has a prologue, headers, etc. that would vary from one request to the next). 

Is there ever a time when a gRPC client would cancel the call but somehow the `GrpcProtocolHandler` would still be used?

My underlying question is, once `callCancelled` has been used below to choose the result status, is there any reason to set `callCancelled` back to `false`? I'm guessing not because it's the `close` method that uses `clientCancelled` but I just wanted to verify that.

(Same question about `headersSent`; would that _ever_ need to be set to `false` again? Again, I assume not but I'm just confirming.)

Reply:
Instances wouldn't be re-used, the underlying HTTP/2 stream is removed after that close method.